1041 feature suggested articles akshanjay

FireBoyAJ24 opened the pull request:
## What problem does this PR solve?

#1041 


## How did you fix the problem?

1. I have finished adding suggested articles sidebar.
2. The user will see a maximum of 6 recent articles from the section the reader is reading in.
3. I have added the functions needed to get the suggested articles and finished the remaining HTML to make the article page more presentable.
4. The format of the suggested articles now comes from @ihciem. After reviewing her work I have added the condition, that if the article page is in the guide then no featured suggested articles are presented.
5. The suggested articles will first come from the category that the host article is from. If there is no articles then the recent articles from the section is presented. Instead of @ihciem 4 articles, I have increased it to 6 articles.

FireBoyAJ24 committed:
Add featured articles (raw) onto the sidebar

FireBoyAJ24 committed:
The get_suggested_articles are now functioning

FireBoyAJ24 committed:
Made the sidebar more presentable

FireBoyAJ24 committed:
Sidebar will show either category or current artic

FireBoyAJ24 committed:
Removed an unneccesary function

SamuelmdLow committed:
Started a visual essay tempalte for articles

SamuelmdLow committed:
Scaled the visual essay template for mobile

SamuelmdLow committed:
Added an introducing block before the split into two columns

SamuelmdLow committed:
Allowed foreword to be multiple blocks long

FireBoyAJ24 committed:
Do not add feaured artices on the guide article pages

FireBoyAJ24 committed:
Add ichiem suggested aricle html instead

SamuelmdLow committed:
Bettered styled and separated js and css

SamuelmdLow committed:
fixed the id of content container

FireBoyAJ24 committed:
Merge branch 'visual-essay-layout-sam-low' into 1041-feature-suggested-articles-Akshanjay

FireBoyAJ24 committed:
Revert "fixed the id of content container"

This reverts commit fedaee8af99876b71f5e11abab32983d79188e9c.

FireBoyAJ24 committed:
Added the suggested featured articles to the wide articles

FireBoyAJ24 committed:
Changing how he suggested articles are displayed on wide articles

SamuelmdLow committed:
Fixed visual essay styling

FireBoyAJ24 committed:
Work in progress

SamuelmdLow committed:
Created visual essay block

SamuelmdLow committed:
Refined style, remade tempalte, made special nav

SamuelmdLow committed:
Separated visual essay block into blocks.py file

FireBoyAJ24 committed:
Mobile Structure is set

FireBoyAJ24 committed:
Added a different featured articles structure for the wide articles

SamuelmdLow committed:
Merge branch 'visual-essay-layout-sam-low' of https://github.com/ubyssey/ubyssey.ca into 1041-feature-suggested-articles-Akshanjay

SamuelmdLow reviewed (changes requested):
The suggested articles list should be be its own template and placed in the article with `{% include` That way it will be easier to edit independently and move around. Also since it is exactly the same as the sidebar in the home page I think you should make a new `sidebar` app, move the sidebars templates and blocks to that, and then reuse that template for the suggested articles.

Also since this pulled from visual-essay, I fetched and merged the updated visual essay branch

FireBoyAJ24 commented:
> The suggested articles list should be be its own template and placed in the article with `{% include` That way it will be easier to edit independently and move around. Also since it is exactly the same as the sidebar in the home page I think you should make a new `sidebar` app, move the sidebars templates and blocks to that, and then reuse that template for the suggested articles.
> 
> 
> 
> Also since this pulled from visual-essay, I fetched and merged the updated visual essay branch


Having a separate sidebar app would allow for abstraction. However, with the different ways at which the sidebar is presented in each type of article page, it makes it difficult to make the abstraction.

I will further investigate the issue and make amendments accordingly.

FireBoyAJ24 committed:
Separate suggested article for article and fw temp

FireBoyAJ24 committed:
Added the current article from the suggested articles from the list

FireBoyAJ24 committed:
Merge branch '1041-feature-suggested-articles-Akshanjay' of https://github.com/ubyssey/ubyssey.ca into 1041-feature-suggested-articles-Akshanjay

FireBoyAJ24 committed:
The html of the magazine suggested article set not css

FireBoyAJ24 committed:
Went back to the last stable visual essay code
Then adjusted the method at which the same article on the same page is identified and removed

FireBoyAJ24 committed:
Merge with the branch to get a stable version

FireBoyAJ24 committed:
Merge branch '1041-feature-suggested-articles-Akshanjay' of https://github.com/ubyssey/ubyssey.ca into 1041-feature-suggested-articles-Akshanjay

FireBoyAJ24 committed:
Made two templates for the suggested articles

SamuelmdLow committed:
Moved logic back to model and simplified templates

SamuelmdLow committed:
Merge branch '1041-feature-suggested-articles-Akshanjay' of https://github.com/ubyssey/ubyssey.ca into 1041-feature-suggested-articles-Akshanjay

SamuelmdLow committed:
Fixed weird things that happened in the last comit

SamuelmdLow commented:
I moved the logic out of the template and simplified a few things

FireBoyAJ24 committed:
Merge branch 'develop' into 1041-feature-suggested-articles-Akshanjay
